modal/_object.py

```diff
@@ -258,24 +258,15 @@ async def hydrate(self, client: Optional[_Client] = None) -> Self:
 
         *Added in v0.72.39*: This method replaces the deprecated `.resolve()` method.
         """
-        if self._is_hydrated:
-            if self.client._snapshotted and not self._is_rehydrated:
-                # memory snapshots capture references which must be rehydrated
-                # on restore to handle staleness.
-                logger.debug(f"rehydrating {self} after snapshot")
-                self._is_hydrated = False  # un-hydrate and re-resolve
+        if not self._is_hydrated:
```

Comment:
Oooh, I like this simplification! Maybe run the full integration test suite before mergin though 😬

Reply:
Had to roll this back as it broke one of the client unit tests. The current diff passes a number of relevant integration test modules (though I haven't run against all of them 😶‍🌫️)

Reply:
Reapplied it to debug and it seems to be passing tests now... Weird!